src/test/java/com/loofah/graph/api/services/SkillServiceTest.java

```diff
@@ -0,0 +1,61 @@
+package com.loofah.graph.api.services;
+
+import com.loofah.graph.api.models.database.Skill;
+import com.loofah.graph.api.models.filters.SkillFilter;
+import com.loofah.graph.api.repositories.SkillRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+
+import java.util.Optional;
+
+import static com.loofah.graph.api.helpers.TestHelpers.GRADE_ID;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SkillServiceTest {
+
+    @Mock
+    private SkillRepository skillRepository;
+
+    @InjectMocks
+    private SkillService skillService;
+
+    @Test
+    public void getById() {
+
+        Skill expectedSkill = Skill.builder().build();
+        when(skillRepository.findById("id")).thenReturn(Optional.of(expectedSkill));
+
+        Skill actualSkill = skillService.getById("id");
+
+        assertEquals(expectedSkill, actualSkill);
+    }
+
+    @Test
+    public void getWithFilter_categoryId() {
+
+        SkillFilter skillFilter = SkillFilter.builder().withGradeId(GRADE_ID).build();
+
+        Query expectedQuery = new Query();
+        expectedQuery.addCriteria(Criteria.where("gradeId").is(GRADE_ID));
+
+        skillService.getWithFilter(skillFilter);
+        verify(skillRepository).findWithQuery(expectedQuery, Skill.class);
+    }
+
+    @Test
+    public void getWithFilter_gradeId() {
```

Comment:
Did you intend to leave these empty?

Reply:
+1 - I also think it would be good to have tests for all the filters (grade/category/craft), and for no filters (or a null filter object).